code/modules/wiremod/components/hud/click_intercept.dm

```diff
@@ -0,0 +1,64 @@
+/**
+ * # Click Intercept Component
+ *
+ * When activated intercepts next click and outputs clicked atom.
+ * Requires a BCI shell.
+ */
+
+/obj/item/circuit_component/click_intercept
+	display_name = "Click Intercept"
+	display_desc = "A component that when activated will intercept next user's click and output the object that was clicked. Requires a shell."
+
+	required_shells = list(/obj/item/organ/cyberimp/bci)
+
+	var/datum/port/output/clicked_atom
+
+	var/obj/item/organ/cyberimp/bci/bci
+	var/intercept_cooldown = 1 SECONDS
+
+/obj/item/circuit_component/click_intercept/Initialize()
+	. = ..()
+	trigger_input = add_input_port("Activate", PORT_TYPE_SIGNAL)
+	trigger_output = add_output_port("Clicked", PORT_TYPE_SIGNAL)
+	clicked_atom = add_output_port("Clicked Object", PORT_TYPE_ATOM)
+
+/obj/item/circuit_component/click_intercept/register_shell(atom/movable/shell)
+	bci = shell
+	RegisterSignal(shell, COMSIG_ORGAN_REMOVED, .proc/on_organ_removed)
+
+/obj/item/circuit_component/click_intercept/unregister_shell(atom/movable/shell)
+	bci = null
+	UnregisterSignal(shell, COMSIG_ORGAN_REMOVED)
+
+/obj/item/circuit_component/click_intercept/input_received(datum/port/input/port)
+	. = ..()
+
+	if(. || !bci)
+		return
+
+	var/mob/living/owner = bci.owner
+	if(!owner || !istype(owner) || !owner.client)
+		return
+
+	if(TIMER_COOLDOWN_CHECK(parent, COOLDOWN_CIRCUIT_CLICK_INTERCEPT))
+		return
+
+	to_chat(owner, "<B>Left-click to activate click interceptor!</B>")
+	owner.client.click_intercept = src
```

Comment:
This is going to be a problem.

Because click interceptors is a *variable*, and not say, a stack, this means that anything that adds its own is going to null this out, requiring it to be reactivated.

For instance--using a spell then cancelling it will not preserve this.

Reply:
Yeah I know but it's not really going to break it.